snap: rely on fonts from the snap

Saviq opened the pull request:
Rather than looking at system-wide fonts, just use the internal ones.

This has the disadvantage of not supporting fonts different than the
preinstalled ones, but system-files feels like the wrong solution
for that either way.

On Classic, access to the system snaps can be provided by the `desktop`
interface.

Saviq committed:
snap: add fonts interface

AlanGriffiths commented on snap/snapcraft.yaml:
```diff
@@ -65,9 +65,12 @@ apps:
       - login-session-control
-      - x11
       - hardware-observe
+      - opengl
```
Do we need to plug opengl? Does that give us anything the wayland slot does not?

Saviq replied:
Oh yeah, wayland doesn't give you access to hardware.

Either way, it's just placating snapcraft, as it was already here:

https://github.com/MirServer/confined-shell-wip/blob/edd36d6596e46c647e95daed022a7ace28cdd814/snap/snapcraft.yaml#L205

But:

```plain
$ snapcraft
Warning: implicit plug assignment in 'desktop-launch' and 'opengl'. Plugs should be assigned to the app to which they apply, and not implicitly assigned via the global 'plugs:' stanza which is intended for configuration only.
(Reference: https://snapcraft.io/docs/snapcraft-interfaces)
```

Saviq replied:
I've now dropped them from the global `plugs:`.

Saviq committed:
snap: avoid implicit plugs

Saviq commented on snap/local/plugs.json:
```diff
@@ -1,5 +1,8 @@
 {
   "desktop-launch":{
     "allow-installation": "true"
+  },
+  "system-files":{
+    "allow-installation": "true"
```
Also added this so that snap-review is happy.

Saviq reviewed (commented):
Hmm not sure why snapcraft considers "will need manual review" as a failure :/

AlanGriffiths commented on snap/snapcraft.yaml:
```diff
@@ -198,12 +201,16 @@ parts:
     default-provider: mesa-core22
+  fonts:
+    interface: system-files
+    read:
+      - /etc/fonts
+      - /usr/share/fonts
+      - /usr/share/fonts/truetype
```
I think we will also need "/usr/share/fonts/TTF" for Arch. (C.f. https://github.com/MirServer/mir/blob/19a573fcb5efd50f1c7e449c76973fb69cf106fe/src/server/shell/decoration/renderer.cpp#L90)

Saviq replied:
Having slept on it… shouldn't we just use everything from inside the snap? And avoid the privileged `system-files` interface altogether?

AlanGriffiths replied:
Yes, I would prefer that

Saviq replied:
> Yes, I would prefer that

Done.

Saviq committed:
snap: rely on fonts from the snap

Rather than looking at system-wide fonts, just use the internal ones.

This has the disadvantage of not supporting fonts different than the
preinstalled ones, but system-files feels like the wrong solution
for that either way.

On Classic, access to the system snaps can be provided by the `desktop`
interface.

AlanGriffiths reviewed (approved):
OK
